modules/barcode/src/barcode.cpp

```diff
@@ -0,0 +1,293 @@
+    return decoders;
+}
+
+class BarDecode
+{
+public:
+    void init(const vector<Mat> &bar_imgs_);
+
+    const vector<Result> &getDecodeInformation()
+    { return result_info; }
+
+    bool decodeMultiplyProcess();
+
+private:
+    vector<Mat> bar_imgs;
+    vector<Result> result_info;
+};
+
+void BarDecode::init(const vector<Mat> &bar_imgs_)
+{
+    bar_imgs = bar_imgs_;
+}
+
+bool BarDecode::decodeMultiplyProcess()
+{
+    class ParallelBarCodeDecodeProcess : public ParallelLoopBody
+    {
+    public:
+        ParallelBarCodeDecodeProcess(vector<Mat> &bar_imgs_, vector<Result> &decoded_info_) : bar_imgs(bar_imgs_),
+                                                                                              decoded_info(
+                                                                                                      decoded_info_)
+        {}
+
+        void operator()(const Range &range) const CV_OVERRIDE
+        {
+            for (int i = range.start; i < range.end; i++)
+            {
+                Mat bin_bar;
+                Result max_res;
+                float max_conf = -1.f;
+                bool decoded = false;
+                for (const auto &decoder:getDecoders())
+                {
+                    if (decoded)
+                    { break; }
+                    for (const auto binary_type : binary_types)
+                    {
+                        binarize(bar_imgs[i], bin_bar, binary_type);
+                        auto cur_res = decoder->decodeROI(bin_bar);
+                        if (cur_res.second > max_conf)
+                        {
+                            max_res = cur_res.first;
+                            max_conf = cur_res.second;
+                            if (max_conf > 0.6)
+                            {
+                                // code decoded
+                                decoded = true;
+                                break;
+                            }
+                        }
+
+                    }
+                }
+
+                decoded_info[i] = max_res;
+            }
+        }
+
+    private:
+        vector<Mat> bar_imgs;
+        vector<Result> &decoded_info;
+    };
+    result_info.clear();
+    result_info.resize(bar_imgs.size());
+
+    ParallelBarCodeDecodeProcess parallelDecodeProcess{bar_imgs, result_info};
+    parallel_for_(Range(0, int(bar_imgs.size())), parallelDecodeProcess);
+    return !result_info.empty();
+}
+
+
+struct BarcodeDetector::Impl
+{
+public:
+    Impl() = default;;
+
+    ~Impl() = default;;
+
+    vector<Mat> initDecode(const Mat &src, const vector<Point2f> &points) const;
+
+    std::shared_ptr<SuperScale> sr;
+    bool use_nn_sr = false;
+};
+
+// return cropped and scaled bar img
+vector<Mat> BarcodeDetector::Impl::initDecode(const Mat &src, const vector<Point2f> &points) const
+{
+    vector<vector<Point2f>> src_points;
+    //CV_TRACE_FUNCTION();
```

Comment:
> `//CV_TRACE_FUNCTION();`

is it needed or not?

If yes, then it should be the first statement of the function/method.